File: src/main/java/com/heybys/oddments/fooddelivery/persistence/shop/MenuJpaRepository.java
package com.heybys.oddments.fooddelivery.persistence.shop;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

import com.heybys.oddments.fooddelivery.domain.shop.Menu;
import com.heybys.oddments.fooddelivery.domain.shop.MenuId;
import com.heybys.oddments.fooddelivery.domain.shop.ShopId;

interface MenuJpaRepository extends JpaRepository<Menu, MenuId> {

    List<Menu> findByShopIdAndOpenIsTrue(ShopId shopId);

    List<Menu> findByShopId(ShopId shopId);
}

File: src/main/kotlin/com/heybys/oddments/fooddelivery/persistence/shop/MenuJpaRepository.kt
package com.heybys.oddments.fooddelivery.persistence.shop

import com.heybys.oddments.fooddelivery.domain.shop.Menu
import com.heybys.oddments.fooddelivery.domain.shop.MenuId
import com.heybys.oddments.fooddelivery.domain.shop.ShopId
import org.springframework.data.jpa.repository.JpaRepository

internal interface MenuJpaRepository : JpaRepository<Menu, MenuId> {

    fun findByShopIdAndIsOpenIsTrue(shopId: ShopId): List<Menu>

    fun findByShopId(shopId: ShopId): List<Menu>
}
